### PR TITLE
mm/alloc: remove all unnecessary cast for alloc

### PR DESCRIPTION
**Pull request by anchao**
## Summary

mm/alloc: remove all unnecessary cast for alloc

Fix the minor style issue and remove unnecessary cast

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]
please add a commit message

**Comment by anchao** [comment]
> please add a commit message

Done

File: arch/arm/src/cxd56xx/cxd56_usbdev.c [hunk 1]
```diff
@@ -3394,7 +3394,7 @@ static int cxd56_usbdev_open(struct file *filep, const char *relpath,
 
   /* Allocate the open file structure */
 
-  priv = (struct cxd56_usbdev_file_s *)kmm_zalloc(
+  priv = kmm_zalloc(
```
[the remark this answers is not in the document]
Comment (anchao) [reply arch/arm/src/cxd56xx/cxd56_usbdev.c:3396]: Done

File: arch/arm/src/cxd56xx/cxd56_usbdev.c [hunk 2]
```diff
@@ -3489,7 +3489,7 @@ static int cxd56_usbdev_dup(const struct file *oldp,
 
   /* Allocate a new container to hold the task and attribute selection */
 
-  newattr = (struct cxd56_usbdev_file_s *)kmm_malloc(
+  newattr = kmm_malloc(
```
[the remark this answers is not in the document]
Comment (anchao) [reply arch/arm/src/cxd56xx/cxd56_usbdev.c]: Done

File: arch/arm/src/imxrt/imxrt_ehci.c [hunk 3]
```diff
@@ -5059,7 +5059,7 @@ struct usbhost_connection_s *imxrt_ehci_initialize(int controller)
   /* Allocate a pool of free Queue Head (QH) structures */
 
   g_qhpool =
-    (struct imxrt_qh_s *)kmm_memalign(32,
+    kmm_memalign(32,
```
[the remark this answers is not in the document]
Comment (anchao) [reply arch/arm/src/imxrt/imxrt_ehci.c]: Done

File: arch/arm/src/imxrt/imxrt_ehci.c [hunk 4]
```diff
@@ -5082,7 +5082,7 @@ struct usbhost_connection_s *imxrt_ehci_initialize(int controller)
   /* Allocate a pool of free Transfer Descriptor (qTD) structures */
 
   g_qtdpool =
-    (struct imxrt_qtd_s *)kmm_memalign(32,
+    kmm_memalign(32,
```
[the remark this answers is not in the document]
Comment (anchao) [reply arch/arm/src/imxrt/imxrt_ehci.c]: Done

File: arch/arm/src/kinetis/kinetis_usbhshost.c [hunk 5]
```diff
@@ -5129,7 +5129,7 @@ struct usbhost_connection_s *kinetis_ehci_initialize(int controller)
   /* Allocate a pool of free Queue Head (QH) structures */
 
   g_qhpool =
-    (struct kinetis_qh_s *)kmm_memalign(32,
+    kmm_memalign(32,
```
[the remark this answers is not in the document]
Comment (anchao) [reply arch/arm/src/kinetis/kinetis_usbhshost.c]: Done

File: arch/arm/src/kinetis/kinetis_usbhshost.c [hunk 6]
```diff
@@ -5152,7 +5152,7 @@ struct usbhost_connection_s *kinetis_ehci_initialize(int controller)
   /* Allocate a pool of free Transfer Descriptor (qTD) structures */
 
   g_qtdpool =
-    (struct kinetis_qtd_s *)kmm_memalign(32,
+    kmm_memalign(32,
```
[the remark this answers is not in the document]
Comment (anchao) [reply arch/arm/src/kinetis/kinetis_usbhshost.c]: Done

File: arch/arm/src/cxd56xx/cxd56_geofence.c [hunk 7]
```diff
@@ -631,7 +631,7 @@ static int cxd56_geofence_register(const char *devpath)
   struct cxd56_geofence_dev_s *priv;
   int                          ret;
 
-  priv = (struct cxd56_geofence_dev_s *)kmm_zalloc(
+  priv = kmm_zalloc(
     sizeof(struct cxd56_geofence_dev_s));
```
Comment (pkarashchenko) [inline arch/arm/src/cxd56xx/cxd56_geofence.c:635]: can this be merged to one line?

File: arch/arm/src/cxd56xx/cxd56_hostif.c [hunk 8]
```diff
@@ -393,7 +393,7 @@ static int hif_initialize(struct hostif_buff_s *buffer)
   /* Setup driver structure */
 
   drv->dev =
-    (struct cxd56_hifdev_s *)kmm_malloc(sizeof(struct cxd56_hifdev_s) * num);
+    kmm_malloc(sizeof(struct cxd56_hifdev_s) * num);
```
Comment (pkarashchenko) [inline arch/arm/src/cxd56xx/cxd56_hostif.c:396]: can this be merged to a one line?

File: arch/risc-v/src/bl602/bl602_oneshot_lowerhalf.c [hunk 9]
```diff
@@ -366,7 +366,7 @@ struct oneshot_lowerhalf_s *oneshot_initialize(int      chan,
 
   /* Allocate an instance of the lower half driver */
 
-  priv = (struct bl602_oneshot_lowerhalf_s *)kmm_zalloc(
+  priv = kmm_zalloc(
     sizeof(struct bl602_oneshot_lowerhalf_s));
```
Comment (pkarashchenko) [inline arch/risc-v/src/bl602/bl602_oneshot_lowerhalf.c:370]: can this be merged to a one line?

File: arch/risc-v/src/common/riscv_addrenv_kstack.c [hunk 10]
```diff
@@ -65,7 +65,7 @@ int up_addrenv_kstackalloc(struct tcb_s *tcb)
 
   /* Allocate the kernel stack */
 
-  tcb->xcp.kstack = (uintptr_t *)kmm_memalign(STACK_ALIGNMENT,
+  tcb->xcp.kstack = kmm_memalign(STACK_ALIGNMENT,
                                               ARCH_KERNEL_STACKSIZE);
```
Comment (pkarashchenko) [inline arch/risc-v/src/common/riscv_addrenv_kstack.c:69]:
```suggestion
  tcb->xcp.kstack = kmm_memalign(STACK_ALIGNMENT,
                                 ARCH_KERNEL_STACKSIZE);
```

File: arch/risc-v/src/esp32c3/esp32c3_oneshot_lowerhalf.c [hunk 11]
```diff
@@ -348,7 +348,7 @@ struct oneshot_lowerhalf_s *oneshot_initialize(int chan,
 
   /* Allocate an instance of the lower half driver */
 
-  priv = (struct esp32c3_oneshot_lowerhalf_s *)kmm_zalloc(
+  priv = kmm_zalloc(
           sizeof(struct esp32c3_oneshot_lowerhalf_s));
```
Comment (pkarashchenko) [inline arch/risc-v/src/esp32c3/esp32c3_oneshot_lowerhalf.c:352]: can this be merged to a one line?

File: arch/xtensa/src/esp32/esp32_oneshot_lowerhalf.c [hunk 12]
```diff
@@ -348,7 +348,7 @@ struct oneshot_lowerhalf_s *oneshot_initialize(int chan,
 
   /* Allocate an instance of the lower half driver */
 
-  priv = (struct esp32_oneshot_lowerhalf_s *)kmm_zalloc(
+  priv = kmm_zalloc(
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/esp32_oneshot_lowerhalf.c:351]: can this be merged to a one line?

File: arch/xtensa/src/esp32/esp32_wlan.c [hunk 13]
```diff
@@ -416,7 +416,7 @@ static inline struct wlan_pktbuf *wlan_alloc_buffer(struct wlan_priv_s *priv)
       return NULL;
     }
 
-  pktbuf = (struct wlan_pktbuf *)kmm_malloc(
+  pktbuf = kmm_malloc(
             sizeof(struct wlan_pktbuf) + WLAN_BUF_SIZE);
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/esp32_wlan.c:420]: can this be merged to a one line?

File: arch/xtensa/src/esp32s2/esp32s2_oneshot_lowerhalf.c [hunk 14]
```diff
@@ -348,7 +348,7 @@ struct oneshot_lowerhalf_s *oneshot_initialize(int chan,
 
   /* Allocate an instance of the lower half driver */
 
-  priv = (struct esp32s2_oneshot_lowerhalf_s *)kmm_zalloc(
+  priv = kmm_zalloc(
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32s2/esp32s2_oneshot_lowerhalf.c:351]: can this be merged to a one line?

File: arch/xtensa/src/esp32s3/esp32s3_oneshot_lowerhalf.c [hunk 15]
```diff
@@ -344,7 +344,7 @@ struct oneshot_lowerhalf_s *oneshot_initialize(int chan, uint16_t resolution)
 
   /* Allocate an instance of the lower-half driver */
 
-  priv = (struct esp32s3_oneshot_lowerhalf_s *)kmm_zalloc(
+  priv = kmm_zalloc(
           sizeof(struct esp32s3_oneshot_lowerhalf_s));
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32s3/esp32s3_oneshot_lowerhalf.c:348]: can this be merged to a one line?

File: boards/arm/s32k1xx/rddrone-bms772/src/s32k1xx_smbus_sbd.c [hunk 16]
```diff
@@ -771,7 +771,7 @@ int smbus_sbd_initialize(int minor, struct i2c_slave_s *i2c_slave_dev)
   /* Allocate an SMBus Smart Battery Data slave device structure */
 
   smbus_sbd_dev =
-    (struct smbus_sbd_dev_s *)kmm_zalloc(sizeof(struct smbus_sbd_dev_s));
+    kmm_zalloc(sizeof(struct smbus_sbd_dev_s));
```
Comment (pkarashchenko) [inline boards/arm/s32k1xx/rddrone-bms772/src/s32k1xx_smbus_sbd.c:774]: can this be merged to a one line?

File: boards/arm/s32k1xx/rddrone-bms772/src/s32k1xx_smbus_sbd.c [hunk 17]
```diff
@@ -802,7 +802,7 @@ int smbus_sbd_initialize(int minor, struct i2c_slave_s *i2c_slave_dev)
   /* Allocate the SMBus Smart Battery Data slave data structure */
 
   smbus_sbd_dev->data =
-    (struct smbus_sbd_data_s *)kmm_zalloc(sizeof(struct smbus_sbd_data_s));
+    kmm_zalloc(sizeof(struct smbus_sbd_data_s));
```
Comment (pkarashchenko) [inline boards/arm/s32k1xx/rddrone-bms772/src/s32k1xx_smbus_sbd.c:805]: can this be merged to a one line?

File: drivers/analog/ads1242.c [hunk 18]
```diff
@@ -585,7 +585,7 @@ int ads1242_register(FAR const char *devpath, FAR struct spi_dev_s *spi,
   /* Initialize the ADS1242 device structure */
 
   priv =
-    (FAR struct ads1242_dev_s *)kmm_malloc(sizeof(struct ads1242_dev_s));
+    kmm_malloc(sizeof(struct ads1242_dev_s));
```
Comment (pkarashchenko) [inline drivers/analog/ads1242.c:588]: can this be merged to a one line?

File: drivers/analog/ltc1867l.c [hunk 19]
```diff
@@ -319,7 +319,7 @@ int ltc1867l_register(FAR const char *devpath, FAR struct spi_dev_s *spi,
   /* Initialize the LTC1867L device structure */
 
   adcpriv =
-    (FAR struct ltc1867l_dev_s *)kmm_malloc(sizeof(struct ltc1867l_dev_s));
+    kmm_malloc(sizeof(struct ltc1867l_dev_s));
```
Comment (pkarashchenko) [inline drivers/analog/ltc1867l.c:322]: can this be merged to a one line?

File: drivers/analog/dac7554.c [hunk 20]
```diff
@@ -246,11 +246,11 @@ FAR struct dac_dev_s *dac7554_initialize(FAR struct spi_dev_s *spi,
   /* Initialize the DAC7554 device structure */
 
   priv =
-    (FAR struct dac7554_dev_s *)kmm_malloc(sizeof(struct dac7554_dev_s));
+    kmm_malloc(sizeof(struct dac7554_dev_s));
```
Comment (pkarashchenko) [inline drivers/analog/dac7554.c:249]: can this be merged to a one line?

File: drivers/audio/tone.c [hunk 21]
```diff
@@ -940,7 +940,7 @@ int tone_register(FAR const char *path, FAR struct pwm_lowerhalf_s *tone,
   /* Allocate the upper-half data structure */
 
   upper =
-    (FAR struct tone_upperhalf_s *)kmm_zalloc(
+    kmm_zalloc(
                               sizeof(struct tone_upperhalf_s));
```
Comment (pkarashchenko) [inline drivers/audio/tone.c:944]: can this be merged to a one line?

File: drivers/input/spq10kbd.c [hunk 22]
```diff
@@ -973,7 +973,7 @@ int spq10kbd_register(FAR struct i2c_master_s *i2c,
   DEBUGASSERT(config->attach != NULL && config->enable != NULL &&
               config->clear  != NULL);
 
-  priv = (FAR struct spq10kbd_dev_s *)kmm_zalloc(
+  priv = kmm_zalloc(
     sizeof(struct spq10kbd_dev_s));
```
Comment (pkarashchenko) [inline drivers/input/spq10kbd.c:977]: can this be merged to a one line?

File: drivers/input/stmpe811_base.c [hunk 23]
```diff
@@ -266,7 +266,7 @@ STMPE811_HANDLE stmpe811_instantiate(FAR struct i2c_master_s *dev,
   /* Allocate the device state structure */
 
 #ifdef CONFIG_STMPE811_MULTIPLE
-  priv = (FAR struct stmpe811_dev_s *)kmm_zalloc(
+  priv = kmm_zalloc(
     sizeof(struct stmpe811_dev_s));
```
Comment (pkarashchenko) [inline drivers/input/stmpe811_base.c:270]: can this be merged to a one line?

File: drivers/ioexpander/pca9538.c [hunk 24]
```diff
@@ -908,7 +908,7 @@ FAR struct ioexpander_dev_s *pca9538_initialize
 #ifdef CONFIG_PCA9538_MULTIPLE
   /* Allocate the device state structure */
 
-  pcadev = (FAR struct pca9538_dev_s *)kmm_zalloc
+  pcadev = kmm_zalloc
                                             (sizeof(struct pca9538_dev_s));
```
Comment (pkarashchenko) [inline drivers/ioexpander/pca9538.c:912]: can this be merged to a one line?

File: drivers/lcd/tda19988.c [hunk 25]
```diff
@@ -722,7 +722,7 @@ static int tda19988_fetch_edid(struct tda1988_dev_s *priv)
       int i;
 
       edid_len =  EDID_LENGTH * (blocks + 1);
```
Comment (pkarashchenko) [inline drivers/lcd/tda19988.c:724]:
```suggestion
      edid_len = EDID_LENGTH * (blocks + 1);
```

File: drivers/leds/lp503x.c [hunk 26]
```diff
@@ -925,7 +925,7 @@ int lp503x_register(const char *devpath, struct i2c_master_s *i2c,
   /* Initialize the LP503X device structure */
 
   struct lp503x_dev_s *priv =
-         (struct lp503x_dev_s *)kmm_malloc(sizeof(struct lp503x_dev_s));
+         kmm_malloc(sizeof(struct lp503x_dev_s));
```
Comment (pkarashchenko) [inline drivers/leds/lp503x.c:928]: can this be merged to a one line?

File: drivers/leds/lp503x.c [hunk 27]
```diff
@@ -925,7 +925,7 @@ int lp503x_register(const char *devpath, struct i2c_master_s *i2c,
   /* Initialize the LP503X device structure */
 
   struct lp503x_dev_s *priv =
```
Comment (pkarashchenko) [inline drivers/leds/lp503x.c:927]: for other PR: `FAR` is missing

File: drivers/leds/ncp5623c.c [hunk 28]
```diff
@@ -269,7 +269,7 @@ int ncp5623c_register(FAR const char *devpath, FAR struct i2c_master_s *i2c,
   /* Initialize the NCP5623C device structure */
 
   FAR struct ncp5623c_dev_s *priv =
-    (FAR struct ncp5623c_dev_s *)kmm_malloc(sizeof(struct ncp5623c_dev_s));
+    kmm_malloc(sizeof(struct ncp5623c_dev_s));
```
Comment (pkarashchenko) [inline drivers/leds/ncp5623c.c:272]: can this be merged to a one line?

File: drivers/power/battery/bq27426.c [hunk 29]
```diff
@@ -419,7 +419,7 @@ FAR struct battery_gauge_dev_s *bq27426_initialize(
 
   /* Initialize the bq27426 device structure */
 
-  priv = (FAR struct bq27426_dev_s *)kmm_zalloc(sizeof(
+  priv = kmm_zalloc(sizeof(
                                                       struct bq27426_dev_s));
```
Comment (pkarashchenko) [inline drivers/power/battery/bq27426.c:423]: can this be merged to a one line?

File: drivers/sensors/apds9960.c [hunk 30]
```diff
@@ -1199,7 +1199,7 @@ int apds9960_register(FAR const char *devpath,
   /* Initialize the APDS9960 device structure */
 
   FAR struct apds9960_dev_s *priv =
-    (FAR struct apds9960_dev_s *)kmm_zalloc(sizeof(struct apds9960_dev_s));
+    kmm_zalloc(sizeof(struct apds9960_dev_s));
```
Comment (pkarashchenko) [inline drivers/sensors/apds9960.c:1202]: can this be merged to a one line?

File: drivers/sensors/as726x.c [hunk 31]
```diff
@@ -398,7 +398,7 @@ int as726x_register(FAR const char *devpath, FAR struct i2c_master_s *i2c)
   /* Initialize the AS726X device structure */
 
   FAR struct as726x_dev_s *priv =
-    (FAR struct as726x_dev_s *)kmm_malloc(sizeof(struct as726x_dev_s));
+    kmm_malloc(sizeof(struct as726x_dev_s));
```
Comment (pkarashchenko) [inline drivers/sensors/as726x.c:401]: can this be merged to a one line?

File: drivers/sensors/bh1750fvi.c [hunk 32]
```diff
@@ -372,7 +372,7 @@ int bh1750fvi_register(FAR const char *devpath, FAR struct i2c_master_s *i2c,
   /* Initialize the BH1750FVI device structure */
 
   FAR struct bh1750fvi_dev_s *priv =
-    (FAR struct bh1750fvi_dev_s *)kmm_malloc(sizeof(struct bh1750fvi_dev_s));
+    kmm_malloc(sizeof(struct bh1750fvi_dev_s));
```
Comment (pkarashchenko) [inline drivers/sensors/bh1750fvi.c:375]: can this be merged to a one line?

File: drivers/sensors/mcp9844.c [hunk 33]
```diff
@@ -362,7 +362,7 @@ int mcp9844_register(FAR const char *devpath, FAR struct i2c_master_s *i2c,
   /* Initialize the MCP9844 device structure */
 
   FAR struct mcp9844_dev_s *priv =
-    (FAR struct mcp9844_dev_s *)kmm_malloc(sizeof(struct mcp9844_dev_s));
+    kmm_malloc(sizeof(struct mcp9844_dev_s));
```
Comment (pkarashchenko) [inline drivers/sensors/mcp9844.c:365]: can this be merged to a one line?

File: drivers/sensors/veml6070.c [hunk 34]
```diff
@@ -274,7 +274,7 @@ int veml6070_register(FAR const char *devpath, FAR struct i2c_master_s *i2c,
   /* Initialize the VEML6070 device structure */
 
   FAR struct veml6070_dev_s *priv =
-    (FAR struct veml6070_dev_s *)kmm_malloc(sizeof(struct veml6070_dev_s));
+    kmm_malloc(sizeof(struct veml6070_dev_s));
```
Comment (pkarashchenko) [inline drivers/sensors/veml6070.c:277]: can this be merged to a one line?

File: drivers/usbdev/usbmsc.c [hunk 35]
```diff
@@ -1456,10 +1456,10 @@ int usbmsc_bindlun(FAR void *handle, FAR const char *drvrpath,
   if (!priv->iobuffer)
     {
 #ifdef CONFIG_USBMSC_WRMULTIPLE
-      priv->iobuffer = (FAR uint8_t *)kmm_malloc(geo.geo_sectorsize *
+      priv->iobuffer = kmm_malloc(geo.geo_sectorsize *
                                                  CONFIG_USBMSC_NWRREQS);
```
Comment (pkarashchenko) [inline drivers/usbdev/usbmsc.c:1460]: please align

File: drivers/video/video.c [hunk 36]
```diff
@@ -834,7 +834,7 @@ static int initialize_scene_parameter(FAR video_mng_t *vmng,
                                       video_scene_params_t **vsp)
 {
   video_scene_params_t *sp =
-    (FAR video_scene_params_t *)kmm_malloc(sizeof(video_scene_params_t));
+    kmm_malloc(sizeof(video_scene_params_t));
```
Comment (pkarashchenko) [inline drivers/video/video.c:837]: can this be merged to a one line?

File: drivers/video/video.c [hunk 37]
```diff
@@ -834,7 +834,7 @@ static int initialize_scene_parameter(FAR video_mng_t *vmng,
                                       video_scene_params_t **vsp)
 {
   video_scene_params_t *sp =
```
Comment (pkarashchenko) [inline drivers/video/video.c:836]: `FAR` is missing

File: sched/mqueue/msginternal.c [hunk 38]
```diff
@@ -107,7 +107,7 @@ void nxmsg_initialize(void)
 {
   FAR struct msgbuf_s *msg;
 
-  msg = (FAR struct msgbuf_s *)kmm_malloc(sizeof(*msg) *
+  msg = kmm_malloc(sizeof(*msg) *
                                           CONFIG_PREALLOC_MQ_MSGS);
```
Comment (pkarashchenko) [inline sched/mqueue/msginternal.c:111]: please align

File: sched/signal/sig_initialize.c [hunk 39]
```diff
@@ -128,7 +128,7 @@ static FAR sigq_t *nxsig_alloc_block(sq_queue_t *siglist, uint16_t nsigs,
 
   /* Allocate a block of pending signal actions. */
 
-  sigqalloc = (FAR sigq_t *)kmm_malloc((sizeof(sigq_t)) * nsigs);
+  sigqalloc = kmm_malloc((sizeof(sigq_t)) * nsigs);
```
Comment (pkarashchenko) [inline sched/signal/sig_initialize.c:131]:
```suggestion
  sigqalloc = kmm_malloc(sizeof(sigq_t) * nsigs);
```

File: sched/signal/sig_dispatch.c [hunk 40]
```diff
@@ -170,7 +170,7 @@ static FAR sigpendq_t *nxsig_alloc_pendingsignal(void)
         {
           /* No... Allocate the pending signal */
 
-          sigpend = (FAR sigpendq_t *)kmm_malloc((sizeof (sigpendq_t)));
+          sigpend = kmm_malloc((sizeof (sigpendq_t)));
```
Comment (pkarashchenko) [inline sched/signal/sig_dispatch.c:173]:
```suggestion
          sigpend = kmm_malloc(sizeof(sigpendq_t));
```

File: sched/signal/sig_allocpendingsigaction.c [hunk 41]
```diff
@@ -85,7 +85,7 @@ FAR sigq_t *nxsig_alloc_pendingsigaction(void)
         {
           /* No...Try the resource pool */
 
-          sigq = (FAR sigq_t *)kmm_malloc((sizeof (sigq_t)));
+          sigq = kmm_malloc((sizeof (sigq_t)));
```
Comment (pkarashchenko) [inline sched/signal/sig_allocpendingsigaction.c:88]:
```suggestion
          sigq = kmm_malloc(sizeof(sigq_t));
```

**Comment by anchao** [comment]
@pkarashchenko  all addressed, thank you!

File: arch/arm64/src/common/arm64_fpu.c [hunk 42]
```diff
@@ -171,7 +171,7 @@ static int arm64_fpu_procfs_open(struct file *filep, const char *relpath,
 
   /* Allocate the open file structure */
 
-  priv = (struct arm64_fpu_procfs_file_s *)kmm_zalloc(
+  priv = kmm_zalloc(
     sizeof(struct arm64_fpu_procfs_file_s));
```
Comment (pkarashchenko) [inline arch/arm64/src/common/arm64_fpu.c:175]: can this be merged to a single line?

File: arch/z80/src/z8/z8_i2c.c [hunk 43]
```diff
@@ -638,7 +638,7 @@ FAR struct i2c_master_s *z8_i2cbus_initialize(int port)
 
   /* Now, allocate an I2C instance for this caller */
 
-  i2c = (FAR struct z8_i2cdev_s *)kmm_zalloc(sizeof(FAR struct z8_i2cdev_s));
+  i2c = kmm_zalloc(sizeof(FAR struct z8_i2cdev_s));
```
Comment (pkarashchenko) [inline arch/z80/src/z8/z8_i2c.c:641]:
```suggestion
  i2c = kmm_zalloc(sizeof(struct z8_i2cdev_s));
```

File: audio/audio.c [hunk 44]
```diff
@@ -930,7 +930,7 @@ int audio_register(FAR const char *name, FAR struct audio_lowerhalf_s *dev)
 
   /* Allocate the upper-half data structure */
 
-  upper = (FAR struct audio_upperhalf_s *)kmm_zalloc(
+  upper = kmm_zalloc(
                                            sizeof(struct audio_upperhalf_s));
```
Comment (pkarashchenko) [inline audio/audio.c:934]: can this be merged to a single line?

File: drivers/sensors/vl53l1x.c [hunk 45]
```diff
@@ -1121,7 +1121,7 @@ int vl53l1x_register(FAR const char *devpath, FAR struct i2c_master_s *i2c)
 
   /* Initialize the vl53l1x device structure */
 
-  priv = (FAR struct vl53l1x_dev_s *)kmm_malloc(
+  priv = kmm_malloc(
     sizeof(struct vl53l1x_dev_s));
```
Comment (pkarashchenko) [inline drivers/sensors/vl53l1x.c:1125]: can this be merged to a single line?

File: fs/mount/fs_automount.c [hunk 46]
```diff
@@ -213,7 +213,7 @@ static int automount_open(FAR struct file *filep)
 
   /* Allocate a new open structure */
 
-  opriv = (FAR struct automounter_open_s *)kmm_zalloc(
+  opriv = kmm_zalloc(
       sizeof(struct automounter_open_s));
```
Comment (pkarashchenko) [inline fs/mount/fs_automount.c:217]: can this be merged to a single line?

**Comment by anchao** [comment]
> Please fix the leftovers and we can move to merge

@pkarashchenko Done, please review again.

File: Documentation/contributing/coding_style.rst [hunk 47]
```diff
@@ -199,7 +199,7 @@ may be used in the file provided that it is used consistently.
       malloc(sizeof(some_medium_name_s);
 
     FAR struct some_medium_name_s *ptr =
-      (FAR struct some_medium_name_s *)malloc(sizeof(some_medium_name_s);
+      malloc(sizeof(some_medium_name_s);
```
Comment (xiaoxiang781216) [inline Documentation/contributing/coding_style.rst:202]: merge

File: arch/arm/src/cxd56xx/cxd56_gnss.c [hunk 48]
```diff
@@ -3131,7 +3131,7 @@ static int cxd56_gnss_register(const char *devpath)
     }
   };
 
-  priv = (struct cxd56_gnss_dev_s *)kmm_zalloc(
+  priv = kmm_zalloc(
     sizeof(struct cxd56_gnss_dev_s));
```
Comment (xiaoxiang781216) [inline arch/arm/src/cxd56xx/cxd56_gnss.c:3135]: merge

File: arch/arm/src/cxd56xx/cxd56_textheap.c [hunk 49]
```diff
@@ -50,7 +50,7 @@
 void *up_textheap_memalign(size_t align, size_t size)
 {
   void *ret;
-  ret = (void *)kmm_malloc(size);
+  ret = kmm_malloc(size);
```
Comment (xiaoxiang781216) [inline arch/arm/src/cxd56xx/cxd56_textheap.c:53]: merge

File: arch/arm/src/cxd56xx/cxd56_usbdev.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply arch/arm/src/cxd56xx/cxd56_usbdev.c:3396]: but no change

File: arch/arm/src/imxrt/imxrt_ehci.c [hunk 50]
```diff
@@ -5058,10 +5057,8 @@ struct usbhost_connection_s *imxrt_ehci_initialize(int controller)
 #  ifndef CONFIG_IMXRT_EHCI_PREALLOCATE
   /* Allocate a pool of free Queue Head (QH) structures */
 
-  g_qhpool =
-    (struct imxrt_qh_s *)kmm_memalign(32,
-                                      CONFIG_IMXRT_EHCI_NQHS *
-                                      sizeof(struct imxrt_qh_s));
+  g_qhpool = kmm_memalign(32, CONFIG_IMXRT_EHCI_NQHS *
+                              sizeof(struct imxrt_qh_s));
```
Comment (xiaoxiang781216) [inline arch/arm/src/imxrt/imxrt_ehci.c:5061]: merge or align

File: arch/arm/src/imxrt/imxrt_ehci.c [hunk 51]
```diff
@@ -5081,10 +5078,8 @@ struct usbhost_connection_s *imxrt_ehci_initialize(int controller)
 #  ifndef CONFIG_IMXRT_EHCI_PREALLOCATE
   /* Allocate a pool of free Transfer Descriptor (qTD) structures */
 
-  g_qtdpool =
-    (struct imxrt_qtd_s *)kmm_memalign(32,
-                                       CONFIG_IMXRT_EHCI_NQTDS *
-                                       sizeof(struct imxrt_qtd_s));
+  g_qtdpool = kmm_memalign(32, CONFIG_IMXRT_EHCI_NQTDS *
+                               sizeof(struct imxrt_qtd_s));
```
Comment (xiaoxiang781216) [inline arch/arm/src/imxrt/imxrt_ehci.c:5082]: ditto

File: arch/arm/src/kinetis/kinetis_usbhshost.c [hunk 52]
```diff
@@ -5128,10 +5127,8 @@ struct usbhost_connection_s *kinetis_ehci_initialize(int controller)
 #  ifndef CONFIG_KINETIS_EHCI_PREALLOCATE
   /* Allocate a pool of free Queue Head (QH) structures */
 
-  g_qhpool =
-    (struct kinetis_qh_s *)kmm_memalign(32,
-                                      CONFIG_KINETIS_EHCI_NQHS *
-                                      sizeof(struct kinetis_qh_s));
+  g_qhpool = kmm_memalign(32, CONFIG_KINETIS_EHCI_NQHS *
+                              sizeof(struct kinetis_qh_s));
```
Comment (xiaoxiang781216) [inline arch/arm/src/kinetis/kinetis_usbhshost.c:5131]: ditto

File: arch/arm/src/kinetis/kinetis_usbhshost.c [hunk 53]
```diff
@@ -5151,10 +5148,8 @@ struct usbhost_connection_s *kinetis_ehci_initialize(int controller)
 #  ifndef CONFIG_KINETIS_EHCI_PREALLOCATE
   /* Allocate a pool of free Transfer Descriptor (qTD) structures */
 
-  g_qtdpool =
-    (struct kinetis_qtd_s *)kmm_memalign(32,
-                                       CONFIG_KINETIS_EHCI_NQTDS *
-                                       sizeof(struct kinetis_qtd_s));
+  g_qtdpool = kmm_memalign(32, CONFIG_KINETIS_EHCI_NQTDS *
```
Comment (xiaoxiang781216) [inline arch/arm/src/kinetis/kinetis_usbhshost.c:5151]: ditto

File: arch/arm/src/sama5/sam_flexcom_spi.c [hunk 54]
```diff
@@ -1857,7 +1857,7 @@ struct spi_dev_s *sam_flex_spibus_initialize(int port)
    * chip select structures.
    */
 
-  flex_spics = (struct sam_flex_spics_s *)kmm_zalloc(
+  flex_spics = kmm_zalloc(
                 sizeof(struct sam_flex_spics_s));
```
Comment (xiaoxiang781216) [inline arch/arm/src/sama5/sam_flexcom_spi.c:1861]: ditto

File: arch/risc-v/src/bl602/bl602_os_hal.c [hunk 55]
```diff
@@ -1068,7 +1068,7 @@ int bl_os_timer_start_periodic(void *timerid, long t_sec, long t_nsec)
 void *bl_os_workqueue_create(void)
 {
   struct work_s *work = NULL;
-  work = (struct work_s *)kmm_calloc(1, sizeof(struct work_s));
+  work = kmm_calloc(1, sizeof(struct work_s));
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/bl602/bl602_os_hal.c:1071]: merge

File: arch/xtensa/src/common/xtensa_oneshot.c [hunk 56]
```diff
@@ -183,7 +183,7 @@ struct oneshot_lowerhalf_s *
 xtensa_oneshot_initialize(uint32_t irq, uint32_t freq)
 {
   struct xoneshot_lowerhalf_s *lower =
-      (struct xoneshot_lowerhalf_s *)kmm_zalloc(sizeof(*lower));
+      kmm_zalloc(sizeof(*lower));
```
Comment (xiaoxiang781216) [inline arch/xtensa/src/common/xtensa_oneshot.c:186]: merge

File: arch/xtensa/src/esp32/esp32_himem_chardev.c [hunk 57]
```diff
@@ -297,7 +297,7 @@ int himem_chardev_register(char *name, size_t size)
 {
   int ret = 0;
   struct himem_chardev_s *dev;
-  dev = (struct himem_chardev_s *)kmm_malloc(sizeof(struct himem_chardev_s));
```
Comment (xiaoxiang781216) [inline arch/xtensa/src/esp32/esp32_himem_chardev.c:300]: merge

File: boards/arm/stm32/mikroe-stm32f4/src/stm32_appinit.c [hunk 58]
```diff
@@ -291,7 +291,7 @@ int board_app_initialize(uintptr_t arg)
 #if defined(CONFIG_RAMMTD) && defined(CONFIG_MIKROE_RAMMTD)
     {
       uint8_t *start =
-          (uint8_t *) kmm_malloc(CONFIG_MIKROE_RAMMTD_SIZE * 1024);
+          kmm_malloc(CONFIG_MIKROE_RAMMTD_SIZE * 1024);
```
Comment (xiaoxiang781216) [inline boards/arm/stm32/mikroe-stm32f4/src/stm32_appinit.c:294]: merge

**Review by acassis — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**